core/src/test/scala/unit/kafka/server/ClientQuotaManagerTest.scala

```diff
@@ -501,6 +501,76 @@ class ClientQuotaManagerTest extends BaseClientQuotaManagerTest {
     }
   }
 
+  @Test
+  def testQuotaTypesEnabledUpdatesOnAddAndRemove(): Unit = {
+    val clientQuotaManager = new ClientQuotaManager(config, metrics, QuotaType.CONTROLLER_MUTATION, time, "")
+    try {
+      // Initially, quotaTypesEnabled should be QuotaTypes.NoQuotas and quotasEnabled should be false
+      assertEquals(QuotaTypes.NoQuotas, clientQuotaManager.getQuotaTypesEnabled)
+      assertFalse(clientQuotaManager.quotasEnabled)
+
+      // Add a client-id quota and quotaTypesEnabled should be QuotaTypes.ClientIdQuotaEnabled
+      clientQuotaManager.updateQuota(
+        None,
+        Some(ClientQuotaManager.ClientIdEntity("client1")),
+        Some(new Quota(5, true))
+      )
+      assertEquals(QuotaTypes.ClientIdQuotaEnabled, clientQuotaManager.getQuotaTypesEnabled)
+      assertTrue(clientQuotaManager.quotasEnabled)
+
```

Comment:
could we add another case here where we add another `ClientIdQuota` and show that `getQuotaTypesEnabled` still remains `QuotaTypes.ClientIdQuotaEnabled`?